File: framework-ts/module/devian-domain-game/features/ability/AbilityAffect.ts
import { AFFECT } from '../../Generated/Game.g';
import { AbilityBase } from './AbilityBase';

export class AbilityAffect extends AbilityBase {
    private mTable: AFFECT | null = null;

    get affectId(): string { return this.mTable?.affect_id ?? ''; }
    get nameId(): string { return this.mTable?.name_id ?? ''; }

    init(table: AFFECT): void {
        this.mTable = table;
    }

    clone(): AbilityAffect {
        const c = new AbilityAffect();
        c.mTable = this.mTable;
        c.copyStatsFrom(this);
        return c;
    }
}

File: framework-cs/apps/UnityExample/Assets/Samples/Devian Foundation/0.1.0/GamePackage/Runtime/Ability/AbilityAffect.cs
using Devian.Domain.Game;

namespace Devian
{
    public sealed class AbilityAffect : AbilityBase
    {
        AFFECT mTable = null;

        public string AffectId => mTable?.affect_id ?? string.Empty;
        public string NameId => mTable?.name_id ?? string.Empty;

        public void Init(AFFECT table)
        {
            mTable = table;
        }

        public override AbilityBase Clone()
        {
            var c = new AbilityAffect();
            c.mTable = mTable;
            c.CopyStatsFrom(this);
            return c;
        }
    }
}
